templates/aws-stack.yml

```diff
@@ -665,6 +666,16 @@ Resources:
       SubnetId: !Ref Subnet1
       RouteTableId: !Ref Routes
 
+  # A resource that depends on the leaf nodes of the VPC configuration so that
+  # a strict ordering can be established on teardown.
+  VpcComplete:
+    Type: AWS::CloudFormation::WaitConditionHandle
```

Comment:
We don’t create a wait condition for this handle so it completes immediately, this is just a convenient cloudformation type to use for the 'barrier' resource.